ca/ca.go

```diff
@@ -148,27 +197,62 @@ func NewCertificateAuthorityImpl(
 		})
 	stats.MustRegister(lintErrorCount)
 
+	certProfileCount := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "certificate_profile",
+			Help: "A counter of certificate profiles used for issuance labelled by profile name, hex representation of the profile hash, certificate purpose (precert/cert), and issuance status (success/failure)",
+		}, []string{"profile", "hash", "purpose", "status"})
+	stats.MustRegister(certProfileCount)
+
 	ca = &certificateAuthorityImpl{
-		sa:             sa,
-		pa:             pa,
-		issuers:        issuers,
-		profile:        certificateProfile,
-		validityPeriod: certExpiry,
-		backdate:       certBackdate,
-		prefix:         serialPrefix,
-		maxNames:       maxNames,
-		keyPolicy:      keyPolicy,
-		log:            logger,
-		signatureCount: signatureCount,
-		signErrorCount: signErrorCount,
-		lintErrorCount: lintErrorCount,
-		clk:            clk,
-		ecdsaAllowList: ecdsaAllowList,
+		sa:               sa,
+		pa:               pa,
+		issuers:          issuers,
+		certProfiles:     certProfiles,
+		validityPeriod:   certExpiry,
+		backdate:         certBackdate,
+		prefix:           serialPrefix,
+		maxNames:         maxNames,
+		keyPolicy:        keyPolicy,
+		log:              logger,
+		signatureCount:   signatureCount,
+		signErrorCount:   signErrorCount,
+		lintErrorCount:   lintErrorCount,
+		certProfileCount: certProfileCount,
+		clk:              clk,
+		ecdsaAllowList:   ecdsaAllowList,
 	}
 
 	return ca, nil
 }
 
+// certificateProfileNameExists checks if the given certificate profile name
+// exists in the CA's list of configured certificate profiles. It returns the
+// hash of the profile or an error. The name is guaranteed to be unique, but the
+// profile values may differ between CA nodes and should not be solely relied
+// upon.
+func (ca *certificateAuthorityImpl) certificateProfileNameExists(name string) ([32]byte, error) {
+	profile, ok := ca.certProfiles.byName[name]
+	if !ok {
+		return [32]byte{}, fmt.Errorf("the CA is not capable of using a profile named %s", name)
+	}
+
+	return profile.GetHash(), nil
+}
```

Comment:
The name of this method implies that it would return a bool, but it's actually a getter method. I've also revised the error and doc comment a bit.

```suggestion
// getCertificateProfileHashByName retrieves the hash of a certificate profile
// by its name. It returns an error if no such profile is configured. Note:
// Profile names are unique, but their configurations may differ across CA
// nodes.
func (ca *certificateAuthorityImpl) getCertificateProfileHashByName(name string) ([32]byte, error) {
	profile, ok := ca.certProfiles.byName[name]
	if !ok {
		return [32]byte{}, fmt.Errorf("certificate profile %q is not configured", name)
	}

	return profile.GetHash(), nil
}
```